wlog/levels.go

```diff
@@ -50,3 +51,55 @@ func (l *LogLevel) UnmarshalText(b []byte) error {
 		return fmt.Errorf("invalid log level: %q", string(b))
 	}
 }
+func (l LogLevel) Enabled(other LogLevel) bool {
+	switch l {
+	case DebugLevel:
+		switch other {
+		case DebugLevel, InfoLevel, WarnLevel, ErrorLevel, FatalLevel:
+			return true
+		}
+	case InfoLevel:
+		switch other {
+		case InfoLevel, WarnLevel, ErrorLevel, FatalLevel:
+			return true
+		}
+	case WarnLevel:
+		switch other {
+		case WarnLevel, ErrorLevel, FatalLevel:
+			return true
+		}
+	case ErrorLevel:
+		switch other {
+		case ErrorLevel, FatalLevel:
+			return true
+		}
+	case FatalLevel:
+		switch other {
+		case FatalLevel:
+			return true
+		}
+	}
+	return false
+}
+
+type AtomicLogLevel struct {
+	value atomic.Value
```

Comment:
Flagging one nuance/thing to keep in mind here which is, per the documentation at https://pkg.go.dev/sync/atomic#Value:

> A Value provides an atomic load and store of a consistently typed value. The zero value for a Value returns nil from Load. Once Store has been called, a Value must not be copied.

> A Value must not be copied after first use.

Since `atomic.Value` is stored here directly and `AtomicLogLevel` is embedded directly in most of the logger structs, this means that, with this change, it's not safe to copy logger structs via assignment once their level has been set. However, making this a pointer doesn't really help, since in that case different logger structs will always share the same log level.

I believe this should be safe since it looks like all of the logger structs are generally private and we only return a pointer to them as a `wlog.Logger` interface, which I believe means that there's no way for callers to create copies of these loggers. However, I think it would be a good idea to verify that we're not copying loggers anywhere in the internal code of WGL and to add a comment to this struct that states that it shouldn't be copied because of this caveat of `atomic.Value`.

Reply:
Yeah this was safe because we always use a reference to the outer (logger) struct so the AtomicLogLevel is never copied.

That said, I think this is a good enough flag and easy to mess up down the road that I've converted things to use `*AtomicLogLevel` to avoid foot-shooting in some later refactor.

It is true that loggers (that want independent log levels) should never be shallow-copied. Given that is not possible by a caller (because the structs are private), do you think a comment on all of them would be helpful?

Reply:
Yeah, I'm fine with either a comment on this struct or on each of the fields -- as long as it's mentioned in one place or the other I think that's sufficient.